bodhi/tests/server/consumers/test_masher.py

```diff
@@ -1183,3 +1183,32 @@ def test_testing_request(self, publish):
         self.assertEqual(t.updates, set([]))
         # The update's title should also have been removed from t.state['updates']
         self.assertEqual(t.state['updates'], [])
+
+
+class TestMasherThread_update_comps(unittest.TestCase):
+    """This test class contains tests for the MasherThread.update_comps() method."""
+
+    def setUp(self):
+        self.masher_thread = MasherThread(u'F17', u'stable', [u'bodhi-2.0-1.fc17'],
+                                          u'bowlofeggs', mock.Mock(), mock.Mock(), mock.Mock())
+
+    @mock.patch('bodhi.server.consumers.masher.os.path.exists', mock.Mock(return_value=False))
+    @mock.patch('bodhi.server.consumers.masher.config', {'comps_dir': '/some/path',
+                                                         'comps_url': 'https://example.com/'})
+    @mock.patch('bodhi.server.consumers.masher.util.cmd')
+    def test_comps_no_dir(self, mock_cmd):
+        calls = [
+            mock.call(['git', 'clone', 'https://example.com/', '/some/path'], '/some'),
+            mock.call(['git', 'pull'], '/some/path'),
+            mock.call(['make'], '/some/path'),
+        ]
+        self.masher_thread.update_comps()
+        self.assertEqual(calls, mock_cmd.call_args_list)
```

Comment:
We should add one more test like this one that tests for when ```os.path.exists``` return True. Also, it'd be good to assert that ```os.path.exists()``` was called with the correct path here and in the new test.